x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransformUpdater.java

```diff
@@ -200,12 +207,13 @@ public static void updateTransform(
     private static void validateTransform(
         TransformConfig config,
         Client client,
+        Map<String, String> securityHeaders,
         boolean deferValidation,
         TimeValue timeout,
         ActionListener<Map<String, String>> listener
     ) {
         ClientHelper.executeWithHeadersAsync(
```

Comment:
Use `executeAsyncWithOrigin`. Then none of the other changes in this file are needed.